crates/core/tests/client.rs

```diff
@@ -17,13 +17,40 @@ mod tests {
     use kakarot_rpc_core::models::ConversionError;
     use kakarot_rpc_core::test_utils::constants::EOA_WALLET;
     use kakarot_rpc_core::test_utils::deploy_helpers::{create_raw_ethereum_tx, deploy_kakarot_system};
-    use reth_primitives::{Address, BlockId, Bytes, H256};
+    use reth_primitives::{Address, BlockId, BlockNumberOrTag, Bytes, H256, U256};
     use reth_rpc_types::Log;
     use starknet::core::types::{BlockId as StarknetBlockId, BlockTag, Event, FieldElement};
     use starknet::core::utils::get_selector_from_name;
     use starknet::providers::jsonrpc::HttpTransport;
     use starknet::providers::{JsonRpcClient, Provider};
 
+    #[tokio::test]
+    async fn test_rpc_should_not_raise_when_eoa_not_deployed() {
```

Comment:
might be overkill to have this test, but having it locally was super useful

Reply:
i like! a good reminder that eoas in EVM all have nonce 0 by default, and don't throw when requiring the nonce of a non-existing addr

Reply:
it's not overkill!